File: 181 excersises/179.cpp
#include <iostream>
using namespace std;

int encontrarMaximo(int *arr, int tamaño) {
    if (tamaño == 0) return 0;
    
    int maximo = *arr; 
    int *ptr = arr + 1;
    
    for (int i = 1; i < tamaño; i++) {
        if (*ptr > maximo) {
            maximo = *ptr;
        }
        ptr++;
    }
    
    return maximo;
}

int main() {
    int tamaño;
    cout << "Ingresa el tamaño de la lista: ";
    cin >> tamaño;
    
    int *arreglo = new int[tamaño];
    
    cout << "Ingresa los " << tamaño << " valores:\n";
    for (int i = 0; i < tamaño; i++) {
        cout << "Valores " << i+1 << ": ";
        cin >> arreglo[i];
    }

    int max = encontrarMaximo(arreglo, tamaño);
    cout << "El valor máximo es: " << max << endl;

    delete[] arreglo;
    
    return 0;
}

File: 181 excersises/179.py
def encontrar_maximo(arreglo):
    if not arreglo:
        return None

    maximo = arreglo[0]
    ptr = 1  
    
    while ptr < len(arreglo):
        if arreglo[ptr] > maximo:
            maximo = arreglo[ptr]
        ptr += 1 
    
    return maximo

tamaño = int(input("Ingresa el tamaño de la lista: "))
arreglo = []
print(f"Ingresa los {tamaño} valores:")
for i in range(tamaño):
    elemento = int(input(f"Valores {i+1}: "))
    arreglo.append(elemento)

maximo = encontrar_maximo(arreglo)
print(f"El valor máximo es: {maximo}")